Loop/Managers/DeviceDataManager.swift

```diff
@@ -1241,6 +1241,14 @@ extension DeviceDataManager {
                 return
             }
 
+            guard !self.doseStore.sharingDenied else {
+                // only clear cache since access to health kit is denied
+                insulinDeliveryStore.purgeCachedInsulinDeliveryObjects() { error in
```

Comment:
This does more than just clear HK. It's main function is to clear the doses from DoseStores's CoreData based InsulinDeliveryStore, and clearing HK should be a side effect. I'm guessing if there's a problem resetting when HK sharing is denied, that the solution is going to be in LoopKit.